# Customer Retention & Churn Analysis

## KPI Analysis

The objective of this notebook is to calculate key customer retention and churn metrics that help measure business performance.

The KPIs analyzed include:

1. Total Customers
2. Active Customers
3. Churned Customers
4. Churn Rate
5. Retention Rate
6. Total Revenue
7. Average Revenue Per User (ARPU)
8. Average Monthly Charges
9. Average Customer Lifetime Value (CLTV)
10. Average Customer Tenure

## Step 1: Import Required Libraries

### Purpose

Import libraries required for KPI calculations and business analysis.


In [3]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)

## Step 2: Load Dataset

### Purpose

Load the Telco Customer Churn dataset into a DataFrame for KPI analysis.

In [4]:
df = pd.read_excel("../data/raw/Telco_customer_churn.xlsx")

## Data Type Correction – Total Charges

### Business Issue

The Total Charges column is stored as a text (object) data type instead of a numeric data type.

### Why It Matters

Revenue metrics such as Total Revenue, ARPU, and Customer Lifetime Analysis require numerical calculations.

### Impact

If the column remains as text, revenue-related KPIs cannot be calculated accurately.

### Solution

Convert the Total Charges column to a numeric data type and handle invalid values appropriately.

In [5]:
df['Total Charges'] = pd.to_numeric(
    df['Total Charges'],
    errors='coerce'
)

In [6]:
df['Total Charges'].dtype

dtype('float64')

### Validation Result

The Total Charges column was successfully converted to a numeric data type and is now ready for revenue-related KPI calculations.

## KPI 1: Total Customers

### Business Question
How many unique customers are present in the dataset?

### Why It Matters
This KPI represents the total customer base of the company and serves as the foundation for calculating churn rate, retention rate, and other customer analytics metrics.

### Formula
Count of Unique Customer IDs

In [7]:
# Calculate total unique customers
total_customers = df['CustomerID'].nunique()

print("Total Customers:", total_customers)

Total Customers: 7043


### Observation
The dataset contains 7,043 unique customers.

### Insight
This represents the total customer base available for retention and churn analysis.

### Business Impact
All future KPIs such as Churn Rate, Retention Rate, and Customer Lifetime Value will be calculated using this customer population.

## KPI 2: Active Customers

### Business Question
How many customers are currently retained by the company?

### Why It Matters
Active customers continue generating recurring revenue and contribute to business growth.

### Formula
Count of Customers where Churn Label = 'No'

In [8]:
# Calculate active customers
active_customers = (df['Churn Label'] == 'No').sum()

print("Active Customers:", active_customers)

Active Customers: 5174


### Observation

The company currently retains 5,174 customers.

### Insight

The company retains 5,174 customers, representing 73.46% of the total customer base.

These retained customers form the primary source of recurring revenue.

### Business Impact

Retained customers are typically more profitable than newly acquired customers because acquisition costs have already been incurred.

## KPI 3: Churned Customers

### Business Question
How many customers have stopped using the company's services?

### Why It Matters
Churned customers represent lost revenue and increased acquisition pressure.

### Formula
Count of Customers where Churn Label = 'Yes'

In [9]:
# Calculate churned customers
churned_customers = (df['Churn Label'] == 'Yes').sum()

print("Churned Customers:", churned_customers)

Churned Customers: 1869


### Observation

The company lost 1,869 customers.

### Insight

A total of 1,869 customers have churned, representing more than one-quarter of the customer base.

### Business Impact

Customer churn directly reduces recurring revenue and negatively impacts customer lifetime value.

## KPI 4: Churn Rate

### Business Question
What percentage of customers have left the company?

### Why It Matters
Churn Rate is one of the most important customer health metrics used by businesses.

### Formula

Churn Rate =

(Churned Customers / Total Customers) × 100

In [10]:
# Calculate churn rate
churn_rate = (churned_customers / total_customers) * 100

print(f"Churn Rate: {churn_rate:.2f}%")

Churn Rate: 26.54%


### Observation

The company has a churn rate of 26.54%.

### Insight

Approximately 1 out of every 4 customers has left the company.

### Business Impact

This churn level is relatively high and may negatively affect revenue growth and long-term profitability.

### Recommendation

Investigate churn drivers such as contract type, payment method, service adoption, and customer tenure.

## KPI 5: Retention Rate

### Business Question
What percentage of customers are still retained by the company?

### Why It Matters
Retention Rate measures the company's ability to keep customers over time.

### Formula

Retention Rate =

(Active Customers / Total Customers) × 100

In [11]:
# Calculate retention rate
retention_rate = (active_customers / total_customers) * 100

print(f"Retention Rate: {retention_rate:.2f}%")

Retention Rate: 73.46%


### Observation

The company has a retention rate of 73.46%.

### Insight

Nearly three out of every four customers continue to use the company's services.

### Business Impact

Retention remains stronger than churn, but improvement opportunities still exist.

### Recommendation

Focus on high-risk customer segments to improve overall retention performance.

## KPI 6: Total Revenue

### Business Question

How much total revenue has been generated by all customers?

### Why It Matters

Revenue is one of the most important business performance metrics.

Understanding total revenue helps quantify the financial impact of customer churn.

### Formula

Sum of Total Charges

In [12]:
# Calculate total revenue generated

total_revenue = df['Total Charges'].sum()

print(f"Total Revenue: ${total_revenue:,.2f}")

Total Revenue: $16,056,168.70


### Observation

The company generated total revenue of $16.06 million from 7,043 customers.

### Insight

The customer base contributes substantial revenue, indicating that customer retention plays a critical role in sustaining business performance.

### Business Impact

Even a small increase in churn can result in significant revenue loss because the company's revenue model depends on recurring customer subscriptions.

### Recommendation

Prioritize retention strategies for high-value customers and identify churn drivers to protect future revenue streams.

## KPI 7: Average Revenue Per User (ARPU)

### Business Question

How much revenue does the average customer generate?

### Why It Matters

ARPU is a key performance metric used to evaluate customer value and business performance.

### Formula

ARPU = Total Revenue ÷ Total Customers

In [13]:
# Calculate Average Revenue Per User

arpu = total_revenue / total_customers

print(f"ARPU: ${arpu:,.2f}")

ARPU: $2,279.73


### Observation

The Average Revenue Per User (ARPU) is $2,279.73.

### Insight

On average, each customer contributes approximately $2,280 in revenue to the company. This indicates that customers represent a significant source of value over their lifecycle.

### Business Impact

Since each customer contributes substantial revenue, customer churn can result in meaningful financial losses. Retaining existing customers can significantly improve overall profitability.

### Recommendation

Focus retention efforts on high-value customer segments and identify customers with declining engagement before they churn.

## KPI 8: Average Monthly Charges

### Business Question

What is the average monthly subscription fee paid by customers?

### Why It Matters

Monthly Charges directly influence revenue generation and customer affordability.

### Formula

Average of Monthly Charges

In [14]:
# Calculate average monthly charges

avg_monthly_charges = df['Monthly Charges'].mean()

print(f"Average Monthly Charges: ${avg_monthly_charges:,.2f}")

Average Monthly Charges: $64.76


### Observation

Customers pay an average monthly charge of $64.76.

### Insight

The company's average monthly subscription fee is approximately $65 per customer. This reflects the recurring revenue generated from the customer base.

### Business Impact

Monthly Charges directly influence revenue generation. However, customers paying higher monthly fees may be more sensitive to perceived service quality and pricing.

### Recommendation

Analyze churn behavior across different monthly charge ranges to determine whether pricing contributes to customer attrition.

## KPI 9: Average Customer Lifetime Value (CLTV)

### Business Question

What is the average lifetime value of a customer?

### Why It Matters

CLTV measures the total value a customer contributes throughout their relationship with the company.

### Formula

Average of CLTV

In [15]:
# Calculate average CLTV

avg_cltv = df['CLTV'].mean()

print(f"Average CLTV: ${avg_cltv:,.2f}")

Average CLTV: $4,400.30


### Observation

The average Customer Lifetime Value (CLTV) is $4,400.30.

### Insight

Customers contribute an average lifetime value of approximately $4,400 throughout their relationship with the company.

### Business Impact

Acquiring and retaining customers with high CLTV can significantly increase long-term profitability and business growth.

### Recommendation

Identify customer segments with the highest CLTV and prioritize them through personalized retention and loyalty programs.

## KPI 10: Average Customer Tenure

### Business Question

How long do customers typically stay with the company?

### Why It Matters

Customer tenure is a strong indicator of loyalty and retention.

### Formula

Average of Tenure Months

In [16]:
# Calculate average customer tenure

avg_tenure = df['Tenure Months'].mean()

print(f"Average Tenure: {avg_tenure:.2f} Months")

Average Tenure: 32.37 Months


### Observation

Customers remain with the company for an average of 32.37 months.

### Insight

The average customer relationship lasts approximately 2.7 years, indicating a moderate level of customer loyalty.

### Business Impact

Longer customer tenure generally leads to higher customer lifetime value, lower acquisition costs, and stronger revenue stability.

### Recommendation

Investigate the characteristics of long-tenure customers and replicate those retention drivers across newer customer segments.

## KPI 11: Churned Revenue

### Business Question

How much revenue is associated with customers who have churned?

### Why It Matters

Customer churn does not only reduce the customer base but also impacts revenue generation.

Measuring churned revenue helps quantify the financial impact of customer attrition and highlights the importance of retention initiatives.

### Formula

Churned Revenue = Sum of Total Charges for Churned Customers

### Business Objective

Determine the amount of revenue associated with customers who discontinued services.

### Expected Outcome

Understand the financial impact of customer churn and identify the potential revenue at risk.

### Metric Used

Total Charges of Customers with Churn Label = "Yes"

In [17]:
# Calculate churned revenue

churned_revenue = df.loc[
    df['Churn Label'] == 'Yes',
    'Total Charges'
].sum()

print(f"Churned Revenue: ${churned_revenue:,.2f}")

Churned Revenue: $2,862,926.90


### Observation

The total revenue associated with churned customers is $2.86 million.

### Insight

Although churned customers represent 26.54% of the customer base, they account for approximately $2.86 million in historical revenue.

This highlights the substantial financial impact of customer attrition on the business.

### Business Impact

Customer churn does not only reduce customer count but also results in significant revenue loss.

If similar churn patterns continue, the company risks losing millions of dollars in future recurring revenue.

### Recommendation

Prioritize churn reduction initiatives for high-value customers and customer segments with elevated churn risk.

Retention strategies focused on reducing churn can directly contribute to revenue protection and long-term business growth.

### Additional Analysis

Determine what percentage of total revenue is associated with churned customers.

In [18]:
churned_revenue_pct = (
    churned_revenue / total_revenue
) * 100

print(
    f"Churned Revenue Percentage: {churned_revenue_pct:.2f}%"
)

Churned Revenue Percentage: 17.83%


### Observation

Churned customers account for 17.83% of the company's total revenue.

### Insight

Although 26.54% of customers churned, they represent 17.83% of total historical revenue.

This suggests that churned customers contribute a meaningful portion of business revenue and should not be viewed solely as customer count loss.

### Business Impact

A continuation of current churn patterns could expose the business to substantial revenue risk.

Even small improvements in retention could protect millions of dollars in customer value and recurring revenue.

### Recommendation

Develop proactive retention programs targeting customers with high revenue contribution and elevated churn risk.

Focus on identifying churn drivers before customers leave the company.

# KPI Analysis Summary

## Customer Metrics

- Total Customers: 7,043
- Active Customers: 5,174
- Churned Customers: 1,869

## Retention Metrics

- Churn Rate: 26.54%
- Retention Rate: 73.46%

## Revenue Metrics

- Total Revenue: $16.06 Million
- Churned Revenue: $2.86 Million
- Churned Revenue Percentage: 17.83%
- Average Revenue Per User (ARPU): $2,279.73
- Average Monthly Charges: $64.76

## Customer Value Metrics

- Average CLTV: $4,400.30
- Average Customer Tenure: 32.37 Months

## Key Business Findings

1. Approximately one out of every four customers has churned, indicating a significant retention challenge.

2. The company retains nearly three-fourths of its customer base, providing a strong foundation for future growth.

3. Customers contribute substantial value, with an average CLTV exceeding $4,400.

4. The average customer relationship lasts more than 32 months, suggesting that improving retention can generate meaningful long-term revenue gains.

## Next Steps

The next phase of analysis will focus on identifying the drivers of customer churn through customer segmentation, service analysis, contract analysis, geographic analysis, and customer behavior analysis.